tests/test_dsl.py

```diff
@@ -29,14 +29,41 @@ def test_empty_selector():
 
 def test_random_card_picker():
 	picker = RandomCardPicker()
-	ids = picker.cards
+	ids = picker.find_cards()
 	for id in ids:
 		card = Card(id)
 		assert card.type is not CardType.HERO
 		assert card.type is not CardType.ENCHANTMENT
 		assert card.type is not CardType.HERO_POWER
 
 
+def test_weighted_random_card_picker():
+	game = prepare_game()
+	game.player1.discard_hand()
+	game.player1.give(WISP)
+
+	spells = len(RandomSpell().find_cards())
+	minions = len(RandomMinion().find_cards())
+	weight = 5
+	ratio = spells / (weight * minions)
+	depth = 100
+	margin = 0.15
+
+	picker = RandomCardPicker().copy_with_weighting(1, type=CardType.SPELL).copy_with_weighting(weight, type=CardType.MINION)
```

Comment:
That's too vague to make a useful response to. The API was discussed for hours with JimboHS last night and this is how he wanted it.